Add a bare minimum localhost hosts file and docs

sdodson opened the pull request:
Hopefully this makes things easier for folks new to ansible

sdodson commented:
@spadgett Does this help? we're still no where near where we need to be as far as developer documentation but this should help at least get an environment installed via the installer then we can dive into where to implement the changes you need to make

spadgett commented:
Yes, thank you. I'm going to give this a shot.

mtnbikenc commented on inventory/byo/hosts.localhost:
```diff
@@ -0,0 +1,21 @@
+etcd
+
+[OSEv3:vars]
+ansible_ssh_user=root
```
Change to `ansible_user`.  The `ssh` parts are deprecated.
http://docs.ansible.com/ansible/latest/intro_inventory.html

mtnbikenc replied:
If not using `root`, will need to add
`ansible_become=yes`

mtnbikenc commented on inventory/byo/hosts.localhost:
```diff
@@ -0,0 +1,21 @@
+
+[OSEv3:vars]
+ansible_ssh_user=root
+debug_level=2
```
Is `debug_level` "bare minimum" since it defaults to 2 in the code?

mtnbikenc commented on CONTRIBUTING.md:
```diff
@@ -74,6 +74,25 @@ If you are new to Git, these links might help:
+the config playbook will run faster the second time through it's still going to
+take a very long time. As such, you may wish to run a smaller subsection of the
+installation playbooks. You can for instance run the node, master, or hosted
+playbooks in playbooks/byo/node/config.yml, playbooks/byo/master/config.yml, and
```
Need to add `openshift-` to both `node` and `master`.

mtnbikenc replied:
Remove 'byo'.

mtnbikenc commented on inventory/byo/hosts.localhost:
```diff
@@ -0,0 +1,21 @@
+openshift_release=3.7
+
+[masters]
+localhost
```
Change to:
`localhost ansible_connection=local`
so that ssh auth doesn't have to be configured.

mtnbikenc commented:
Might want to add:
`openshift_disable_check=disk_availability,memory_availability`
so it doesn't complain on smaller hosts.

mtnbikenc commented:
FYI, I ran into some local temp dir funkiness for syncing etcd certs.  I've not looked further into it yet.

mtnbikenc commented:
Need to address a few things before this merges.
/lgtm cancel

michaelgugino commented on inventory/byo/hosts.localhost:
```diff
@@ -0,0 +1,21 @@
+ansible_ssh_user=root
+debug_level=2
+openshift_deployment_type=origin
+openshift_release=3.7
```
Seeing as 3.7 packages aren't widely available yet, we should use 3.6.

sdodson replied:
lets see if I can just make it readily available

michaelgugino commented on inventory/byo/hosts.localhost:
```diff
@@ -0,0 +1,21 @@
+#bare minimum hostfile
```
Fedora won't run with this alone due to python3 requirements.

michaelgugino commented on inventory/byo/hosts.localhost:
```diff
@@ -0,0 +1,21 @@
+localhost
+
+[nodes]
+localhost openshift_schedulable=true openshift_node_labels="{'region': 'primary', 'zone': 'default'}"
```
I don't think registry and router will be scheduable without region: infra, unless you change their selectors.

michaelgugino replied:
I believe labels can take a list.

michaelgugino commented on README.md:
````diff
@@ -67,7 +67,16 @@ Fedora:
+## Simple all-in-one localhost Installation
+This assumes that you've installed the base dependencies and you're running on
+Fedora or RHEL.
+```
````
Don't forget to add prerequisites.yml here :)

mtnbikenc commented on CONTRIBUTING.md:
````diff
@@ -74,6 +74,25 @@ If you are new to Git, these links might help:
+```
+git clone https://github.com/openshift/openshift-ansible
+cd openshift-ansible
+ansible-playbook -i inventory/byo/hosts.localhost playbooks/byo/config.yml
````
Inventory should go in /inventory/hosts.localhost to go along with https://github.com/openshift/openshift-ansible/pull/6399
Should use playbooks/deploy_cluster.yml.

sdodson commented:
I think I've addressed all of the concerns so far, I tried to remove all instances of byo that made sense currently.

When we rip out byo/config.yml we'll need to fix the installer image as it currently calls one playbook and we'd need to amend it to call a list of playbooks.

sdodson commented:
Err, i guess it's broken today, I thought we were going to make playbooks/byo/config.yml call prerequisites first

sdodson commented:
Dropped the byo cleanup infavor of #6399

mtnbikenc commented:
Results from my testing:

1. Inventory file needs to be moved down one level
2. Running as `root` works.  Using another user with `ansible_become=yes` needs further work.
3. I still think it would be a good idea to add `openshift_disable_check=disk_availability,memory_availability` since local installs are likely not going to have the recommended resources.

sdodson commented:
1 & 2 addressed, I'll work through testing non root user.

mtnbikenc commented:
@sdodson After further investigation, I've found that it is expected behavior when running Ansible against localhost and using an unprivileged user with ansible_become=yes, the playbook will fail due to permissions rights.  The only ways to run a localhost install will be either run as root or run ansible-playbook through sudo.  Additionally, in testing, I've found there are many invalid uses of `become: no` on `local_action:` tasks.

See: https://github.com/ansible/ansible/issues/15933#issuecomment-220657744

mtnbikenc commented on inventory/hosts.localhost:
```diff
@@ -0,0 +1,29 @@
+[OSEv3:vars]
+ansible_user=root
+# if ansible_user is not root you must set the following
+#ansible_become=yes
```
Lines 9-11 can be removed due to the requirement of either running as root or running ansible-playbook using sudo.

mtnbikenc commented on README.md:
````diff
@@ -67,7 +67,17 @@ Fedora:
+## Simple all-in-one localhost Installation
+This assumes that you've installed the base dependencies and you're running on
+Fedora or RHEL.
+```
````
Need to add a note about either running as root or using sudo.

mtnbikenc commented:
/lgtm cancel

vrutkovs commented:
Works fine, agreed with Russel's comments

sdodson commented:
Updated per @mtnbikenc suggestions.
